pinot-broker/src/main/java/com/linkedin/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -311,6 +329,68 @@ private void validateRequest(BrokerRequest brokerRequest) {
             "Value for 'LIMIT' (" + limit + ") exceeds maximum allowed value of " + _queryResponseLimit);
       }
     }
+
+    // Checks whether the query contains non-existence columns.
+    // Table name has already been verified before hitting this line.
+    String tableName = brokerRequest.getQuerySource().getTableName();
```

Comment:
These seem like expensive operations being added in the query path. At the very least, this feature should be off by default.

Reply:
+1 for conf control.

Reply:
Done